src/main.ts

```diff
@@ -175,25 +175,31 @@ export default class RecipeGrabber extends Plugin {
 			if (!tags || typeof tags != "string") {
 				return "";
 			}
-			var tagsArray = tags.split(",");
-			var tagString = "";
+			const tagsArray = tags.split(",");
+			let tagString = "";
 			for (const tag of tagsArray) {
 				tagString += "- " + tag.trim() + "\n";
 			}
 			return tagString;
 		});
 
+		// quick function to check if a string is a valid date
+		function isValidDate(d: string): boolean {
+			return !isNaN(Date.parse(d));
+		}
+
 		handlebars.registerHelper("magicTime", function (arg1, arg2) {
 			if (typeof arg1 === "undefined") {
 				// catch undefined / empty
 				return "";
 			}
 			// Handlebars appends an ubject to the arguments
-			if (arguments.length == 1) {
+			if (arguments.length === 1) {
 				// magicTime
 				return dateFormat(new Date(), "yyyy-mm-dd HH:MM");
-			} else if (arguments.length == 2) {
-				if (new Date(arg1) == "Invalid Date") {
+			} else if (arguments.length === 2) {
+				// check if arg1 is a valid date
+				if (isValidDate(arg1)) {
```

Comment:
```suggestion
				if (!isValidDate(arg1)) {
```
If it's not a valid date, check if it's a ISO 8601 duration string (PT1H30M), else assume it's a valid dateFormat mask.

Reply:
Can you add a suggested change to the new version or update the PR? not sure how you'd like it to check.